cancer_ai/validator/model_manager.py

```diff
@@ -23,7 +24,11 @@ def __init__(self, config, db_controller, parent: Optional["CompetitionManager"]
         self.api = HfApi(token=self.config.hf_token)
         self.hotkey_store: dict[str, ModelInfo] = {}
         self.parent = parent
+        self.subtensor = subtensor
 
+        # Default subtensor is not archive, but we need it for fetching the historical extrinsics data
+        if subtensor is not None and "test" not in self.subtensor.chain_endpoint.lower():
+            self.subtensor = bt.subtensor(network="archive")
```

Comment:
czy inicjalizacja robi coś poza tworzeniem obiektu? jeśli tak, to może lepiej tworzyć raz i reużywać?

Reply:
tak robie